File: code-samples/clean-architecture-java/src/main/java/com/example/usecase/GetArticleUseCase.java
package com.example.usecase;

import com.example.entity.Article;
import com.example.entity.ArticleId;

public class GetArticleUseCase {

    private final ArticleGateway articleGateway;

    public GetArticleUseCase(ArticleGateway articleGateway) {
        this.articleGateway = articleGateway;
    }

    public Article execute(ArticleId articleId) {
        return articleGateway.getArticle(articleId);
    }
}

File: code-samples/clean-architecture-kotlin/src/main/kotlin/com/example/usecase/GetArticleUseCase.kt
package com.example.usecase

import com.example.entity.ArticleId

class GetArticleUseCase(
    private val articleGateway: ArticleGateway,
    private val articlePresenter: ArticlePresenter
) {

    fun execute(articleId: ArticleId) {
        val article = articleGateway.getArticle(articleId)
        articlePresenter.showArticle(article)
    }
}
